File: client/apiHelper.go
package main

import (
"fmt"
"bytes"
"net/http"
"./variables"
)

func checkUser(username string) bool{
    var userPresent bool
        
    url := variables.AWSEndPoint + "/checkUser"

    var jsonprep string = "{\"username\":\"" + username + "\"}"
    var jsonStr = []byte(jsonprep)
    
    req, err := http.NewRequest("POST", url, bytes.NewBuffer(jsonStr))
    req.Header.Set("Content-Type", "application/json")

    client := http.Client{}
    resp, err := client.Do(req)

    if err != nil {
        //panic(err)  
    }
    buf := new(bytes.Buffer)
    buf.ReadFrom(resp.Body)
    response := buf.String()

    //fmt.Println(response)
    defer resp.Body.Close()
    userPresent = response

    return userPresent
}



func main() {
    var username,password string
    fmt.Println("Enter your username:")
    fmt.Scanf("%s", &username)
    
    fmt.Println("Enter your password:")
    fmt.Scanf("%s", &password)   

    checkUser(username)
}

File: client/apiHelper.py
import requests, json
import variables, pymongo
from pymongo import MongoClient
from bson.json_util import dumps

client = MongoClient()
db = client['gochat']
user = db['user']
user.ensure_index('username',unique=True)

def checkUser(username):
    url = variables.AWSEndPoint + "/checkUser"
    data = {"username": username}
    json_data = json.dumps(data)
    headers = {'Content-type': 'application/json'}
    response = requests.post(url, data=json_data, headers=headers)
    return response.text

def registerUser(username, password):
    url = variables.AWSEndPoint + "/registerUser"
    data = {"username": username, "password": password}
    json_data = json.dumps(data)
    headers = {'Content-type': 'application/json'}
    response = requests.post(url, data=json_data, headers=headers)
    try:
        db.user.insert_one({'username':username, 'password': hash_pass(password)})
    except:
        print "nononono"

    return response.text

def hash_pass(password):
    # used to hash the password similar to how MySQL hashes passwords with the password() function.
    import hashlib
    hash_password = hashlib.sha1(password.encode('utf-8')).digest()
    hash_password = hashlib.sha1(hash_password).hexdigest()
    hash_password = '*' + hash_password.upper()
    return hash_password